aten/src/ATen/native/mps/operations/Indexing.h

```diff
@@ -0,0 +1,56 @@
+//  Copyright © 2022 Apple Inc.
+
+#include <ATen/ATen.h>
+#include <ATen/Tensor.h>
+#include <ATen/Utils.h>
+#include <ATen/mps/MPSStream.h>
+#include <ATen/native/mps/TensorFactory.h>
+#include <c10/core/ScalarType.h>
+#include <torch/library.h>
+#include <unordered_map>
+
+#ifdef __OBJC__
+#include <MetalPerformanceShaders/MetalPerformanceShaders.h>
+#endif
+
+using namespace at::mps;
+
+namespace at {
+namespace native {
+namespace mps {
+
+std::string getMetalScalarType(ScalarType scalar_type) {
+  std::string res = "";
+  switch (scalar_type) {
+    case ScalarType::Float:
+      res = "float"; break;
+    case ScalarType::Half:
+      res = "half";  break;
+    case ScalarType::Long:
+      res = "long";  break;
+    case ScalarType::Int:
+      res = "int";   break;
+    case ScalarType::Short:
+      res = "short"; break;
+    case ScalarType::Char:
+      res = "char"; break;
+    case ScalarType::Byte:
+      res = "uchar"; break;
+    case ScalarType::Bool:
+      res = "bool";  break;
```

Comment:
You need one kernel per type size, isn't it?
```suggestion
    case ScalarType::Long:
      res = "64bit";  break;
    case ScalarType::Float:
    case ScalarType::Int:
      res = "32bit"; break;
    case ScalarType::Half:
    case ScalarType::Short:    
      res = "16bit";  break;
    case ScalarType::Char:
    case ScalarType::Byte:
    case ScalarType::Bool:
      res = "8bit";  break;
```